icepyx/core/spatial.py

```diff
@@ -32,38 +33,48 @@ def geodataframe(extent_type, spatial_extent, file=False):
     file : boolean, default False
         Indication for whether the spatial_extent string is a filename or coordinate list
 
+    Returns
+    -------
+    gdf : GeoDataFrame
+        Returns a GeoPandas GeoDataFrame containing the spatial extent.
+        The GeoDataFrame will have only one entry unless a geospatial file
+        was submitted.
+
     See Also
     --------
     icepyx.Query
 
     Examples
     --------
     >>> reg_a = ipx.Query('ATL06',[-55, 68, -48, 71],['2019-02-20','2019-02-28'])
-    >>> gdf = geodataframe(reg_a.spatial.extent_type, reg_a.spatial.spatial_extent)
+    >>> gdf = geodataframe(reg_a.spatial.extent_type, reg_a.spatial.extent)
     >>> gdf.geometry
-    0   POLYGON ((-55.00000 68.00000, -55.00000 71.000...
+    0    POLYGON ((-48.00000 68.00000, -48.00000 71.000...
     Name: geometry, dtype: geometry
     """
 
+    if xdateline is not None:
+        xdateline = xdateline
+    elif file:
+        pass
+    else:
+        xdateline = check_dateline(extent_type, spatial_extent)
+    # print("this should cross the dateline:" + str(xdateline))
+
     if extent_type == "bounding_box":
-        boxx = [
-            spatial_extent[0],
-            spatial_extent[0],
-            spatial_extent[2],
-            spatial_extent[2],
-            spatial_extent[0],
-        ]
-        boxy = [
-            spatial_extent[1],
-            spatial_extent[3],
-            spatial_extent[3],
-            spatial_extent[1],
-            spatial_extent[1],
-        ]
-        # DevGoal: check to see that the box is actually correctly constructed;
-        # have not checked actual location of test coordinates
+        if xdateline == True:
+            cartesian_lons = [i if i > 0 else i + 360 for i in spatial_extent[0:-1:2]]
```

Comment:
Trying to wrap my head around this logic. So negative longitude values get 360 added to them (e.g. -1 becomes 359), while positive longitude values remain intact (e.g. +1 stays as +1)?

Shouldn't it be that if any longitude values are negative (in the range -180 to 0), 180 gets added to the coordinates?

Reply:
I definitely had to do a few laps on this one too.

>Shouldn't it be that if any longitude values are negative (in the range -180 to 0), 180 gets added to all of the longitude coordinates?

No, because it's only the -180 to 0 (or 181 to 360) range of coordinates that need to be adjusted. If you added 180 to everything, you'd effectively be shifting the coordinates halfway around the world. So...

>So negative longitude values get 360 added to them (e.g. -1 becomes 359), while positive longitude values remain intact (e.g. +1 stays as +1)?

Exactly. It's counterintuitive, but has to do with the fact that Shapely (and GeoPandas) use a cartesian plane system (but not spherical geometry computations). Thus, in order for it to correctly handle something crossing the dateline, it needs coordinates in 0-360 space or will automagically "correct" the geometry to not cross the dateline.

Reply:
Wait, I'm getting confused. What is 0 in the 0-360 longitude range, is it at Greenwich or at the International date line? Maybe we need to add a proper unit test for this...

Trying to read https://gis.stackexchange.com/questions/226605/shapely-polygons-crossing-the-antimeridian and https://towardsdatascience.com/around-the-world-in-80-lines-crossing-the-antimeridian-with-python-and-shapely-c87c9b6e1513, and I'm wondering if [`shapely.affinity.translate`](https://shapely.readthedocs.io/en/1.8.5.post1/manual.html#shapely.affinity.translate) might be useful to simplify the (-180 to 180) to (0-360) conversion code (i.e. replace the list-comprehension with `shapely.affinity.translate(geom, xoff=180)`. But let's figure out the logic first :slightly_smiling_face:

Reply:
It's crude (remarkably hard to find a map with both sets of coordinates), but visually:
![image](https://user-images.githubusercontent.com/11756442/198088404-a8c891b3-b1b7-47d8-84f3-275ea439ea97.png)

Reply:
> I'm wondering if [`shapely.affinity.translate`](https://shapely.readthedocs.io/en/1.8.5.post1/manual.html#shapely.affinity.translate) might be useful to simplify the (-180 to 180) to (0-360) conversion code (i.e. replace the list-comprehension with `shapely.affinity.translate(geom, xoff=180)`

The problem with `shapely.affinity.translate` is that it requires a valid geometry input, and that's where the issue first pops up. Consider:

```
import shapely.geometry as geom
# across dateline example
bbox = [-179,60,179,62]
b_bbox = [179,-60,-179,-62] # this is "auto-corrected" to match bbox 

for box in [bbox, b_bbox]:
  fmtbox = geom.box(*box)
  print(fmtbox.bounds)
  
  poly = geom.Polygon.from_bounds(*box)
  print(poly)
```

versus: 
```
# 0 to 360 ex
bbox = [181,60,179,62] # this is auto-corrected to match b_bbox
b_bbox = [179,60,181,62] # this properly crosses the dateline
```

Reply:
Ah ok, so 0 in 0-360 is actually at Greenwich, not the International date line, which makes sense for solving the cross-dateline issue now that I think of it (my bias with NZ-centred world maps is confusing me :laughing:). In that case the if-then logic is sound, and `shapely.affinity.translate` won't work since it's not a simple translational shift.Thanks again for the clarification!

Reply:
Thanks for checking the logic! Having to go through and explain it is always a good exercise to make sure we've got it right. : )